### ISSUE TITLE
Make pull request files sections collapsable 

### ISSUE THREAD

**Issue by eliasjpr** [isaacs/github#671]
I think it would be nice and efficient to have the section bar for each  file collapsable, this makes for navigating through the pull request files faster when review and avoiding so much scrolling up and down. 

<img width="1392" alt="screen_shot_2016-05-11_at_4_57_10_pm" src="https://cloud.githubusercontent.com/assets/1685772/15196280/9a7415fc-1799-11e6-986d-56f67f734c98.png">

**Comment by Mottie** [comment]
If you don't mind userscripts, there is one that adds a toggle button:

https://greasyfork.org/en/scripts/18788-github-diff-file-toggle

**Comment by cirosantilli** [comment]
An index would achieve similar functionality: https://github.com/isaacs/github/issues/639

**Comment by eliasjpr** [comment]
@Mottie interesting... mmm is this a browser plugin?

@cirosantilli while the index would achieve similar functionality sometimes people just want to review the pull request file by file and the index feature will require more than necessary back and forth clicks.

**Comment by Mottie** [comment]
Oh, no it's a userscript, not a browser plugin. You will need a userscript browser plugin though. Get [details here](https://github.com/Mottie/GitHub-userscripts#github-userscripts-) on where to find the plugin and userscript.

**Comment by eliasjpr** [comment]
Oh nice! That made me feel that I've been coding in a cave 😆

**Comment by eliasjpr** [comment]
@Mottie While it eases the pain a little still I think having `expandable/collapsible` sections would be more useful

**Comment by Mottie** [comment]
I'm not quite sure what you mean... are you referring to code folding?

**Comment by eliasjpr** [comment]
@Mottie this is what I'm referring to http://getbootstrap.com/javascript/#collapse-example-accordion I think this can give you a better idea of what I meant

**Comment by Mottie** [comment]
That is essentially what the userscript does, just without the animation... I don't plan on adding animation.

**Comment by eliasjpr** [comment]
Yes the script does a similar function except that all everything is collapsed except for the current item, but I guess this kind of feature would need more demand is really a matter of preferences. :)

**Comment by TPS** [comment]
@eliasjpr You probably shouldn't close this until @GitHub implements this natively; reacting w/ 🎉 is probably more apropos.

**Comment by Mottie** [comment]
Oh, you mean start with all but the first collapsed.... I could add an option.

**Comment by Mottie** [comment]
@eliasjpr Please try out the [updated userscript](https://greasyfork.org/en/scripts/18788-github-diff-file-toggle). Enable the "accordion" mode by clicking on the Tampermonkey/Greasmonkey icon and selecting the "GitHub Diff File Toggle" option, and setting the value to "true".

**Comment by aungerh** [comment]
I made a small userscript that adds a button and makes the entire section collapsible

https://gist.github.com/aungerh/e4a014551cf312a9ffbf0b41745a8b68

might be useful to someone
